### PR TITLE
WIP: Radius random generator

### PR DESCRIPTION
**Pull request by hamax97**
* Radius random generator.
* Initialization function moved to its own header and .cpp files.
* Updated Makefile with the new .h and .cpp files.
* Changed from `radious` to `radius` variable.
* Seed for random generator from arguments file.
* README changed describing the seed argument + Typo fixed.

### CODE REVIEW

File: include/config.h [hunk 1]
```diff
@@ -8,7 +8,7 @@ typedef struct {
   double dt;
   int m; // Size in X.
   int n; // Size in Y.
-  double r;
+  int seed; // Random generator seed
```
Comment (BalmungSan) [inline include/config.h]: Please add a dot _(`.`)_ at the end of the comments. :+1:

File: include/data.h [hunk 2]
```diff
@@ -6,7 +6,7 @@
 typedef struct {
   double x_coordinate;
   double y_coordinate;
-  double radious;
+  double radius;
```
Comment (BalmungSan) [inline include/data.h:9]: Ty :+1:

**Review by BalmungSan — DISMISSED** [review]
Cool, thanks.

Just one thing, are you sure this is right `y = particles_last_height[i % max_in_x] + current_radius;`.
Should not it be in the `i + 1`, because it will be for the next step, or am I missing something?

**Comment by hamax97** [comment]
> Cool, thanks.
> 
> Just one thing, are you sure this is right `y = particles_last_height[i % max_in_x] + current_radius;`.
> Should not it be in the `i + 1`, because it will be for the next step, or am I missing something?

The key is that the index starts at 1. The first value of `y` is calculated outside the loop. Then with index equals 1, 1 % 10, for example, gives us 1, which is the position in the array that we want for the next iteration. And so on.

File: include/config.h (hunk 1 above)
Comment (hamax97) [reply include/config.h]: Done!

**Review by BalmungSan — DISMISSED** [review]

**Review by BalmungSan — DISMISSED** [review]

**Comment by hamax97** [comment]
Paraview shows a weird behavior in the matrix of particles even before the falling particle crashes the matrix, they appear and disappear randomly. I suggest to review it with the team before merging.

**Review by BalmungSan — APPROVED** [review]
:+1:

**Pull request merged**